### PR TITLE
Crm 12 entity customers

### PR DESCRIPTION
**Pull request by MateuszPioro**
Create pull request.

### CODE REVIEW

File: backend/src/main/java/com/crm/App.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package com.crm;
 
+
```
Comment (Fellway) [inline backend/src/main/java/com/crm/App.java:3]: It's not needed

File: backend/src/main/java/com/crm/App.java [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
+
```
Comment (Fellway) [inline backend/src/main/java/com/crm/App.java:8]: It's not needed

File: backend/src/main/java/com/crm/model/db/CarEntity.java [hunk 3]
```diff
@@ -40,4 +35,7 @@
     @ManyToOne(cascade = CascadeType.ALL)
     @JoinColumn(name = "type_id")
     private CarTypeEntity carTypeEntity;
-}
+
+    @ManyToMany(mappedBy="cars")
+    Set<CustomerEntity> cars = new HashSet<>();
```
Comment (Fellway) [inline backend/src/main/java/com/crm/model/db/CarEntity.java:40]: private

File: backend/src/main/java/com/crm/model/db/CustomerEntity.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import com.sun.istack.NotNull;
+import lombok.*;
+import org.hibernate.annotations.Cascade;
+
+import javax.persistence.*;
+import java.util.HashSet;
+import java.util.Set;
+
+@Entity
+@Builder
```
Comment (Fellway) [inline backend/src/main/java/com/crm/model/db/CustomerEntity.java:13]: Builder is not needed

File: backend/src/main/java/com/crm/model/db/CustomerEntity.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+@AllArgsConstructor
```
Comment (Fellway) [inline backend/src/main/java/com/crm/model/db/CustomerEntity.java:14]: It's not needed

File: backend/src/main/java/com/crm/model/db/CustomerEntity.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    @NotNull
+    private String address;
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(
+            name = "customers_cars",
+            joinColumns = @JoinColumn(name = "customers_id"),
+            inverseJoinColumns = @JoinColumn(name = "car_id")
+    )
+    Set<CarEntity> cars = new HashSet<>();
```
Comment (Fellway) [inline backend/src/main/java/com/crm/model/db/CustomerEntity.java:43]: private

File: db-migration/pom.xml [hunk 7]
```diff
@@ -36,7 +36,7 @@
                 <configuration>
                     <url>jdbc:postgresql://localhost:5432/crm</url>
                     <user>postgres</user>
-                    <password>postgres</password>
+                    <password>smentek12</password>
```
Comment (Fellway) [inline db-migration/pom.xml:39]: remove this from PR

**Review by Fellway — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/crm/model/db/CustomerEntity.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    private String address;
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(
+            name = "customers_cars",
+            joinColumns = @JoinColumn(name = "customers_id"),
+            inverseJoinColumns = @JoinColumn(name = "car_id")
+    )
+    private Set<CarEntity> cars = new HashSet<>();
+
```
Comment (SteelWro) [inline backend/src/main/java/com/crm/model/db/CustomerEntity.java:42]: Cause we use Collection like Set and other we need to ovveride `equals `and `hashcode `methods

**Review by Fellway — APPROVED** [review]

**Pull request merged**